src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -25,16 +26,71 @@ public boolean isFull() {
 
 	public int top() {
 		if (isEmpty())
-			return -1;
-		return last.value;
+			return LIST_EMPTY;
+		return getLast().getValue();
 	}
 
 	public int pop() {
 		if (isEmpty())
-			return -1;
-		int ret = last.value;
-		last = last.prev;
-		return ret;
+			return LIST_EMPTY;
+		int returnedValue = getLast().getValue();

```

Comment:
This is an exaggeration. Using private fields internally usually do not require encapsulation.